src/attackHandler.ts

```diff
@@ -0,0 +1,48 @@
+import { Character } from './character';
+import { WeaponCollision as collision } from './weaponCollision';
+
+export class AttackHandler {
+  static handleAttack(attacker: Character) {
+    if (attacker.weapon === undefined) {
+      return;
+    }
+    if (Date.now() - attacker.attackTimer >= attacker.weapon.cooldown) {
+      attacker.isAttacking = true;
+      attacker.attackTimer = Date.now();
+    }
+  }
+
+  static update(attacker: Character, defenders: Array<Character>) {
+    if (attacker.weapon === undefined) {
```

Comment:
No need for this check if we have `canAttack`.

Reply:
I kept this check in the combined function. Seems to still be relevant.